pandas/core/missing.py

```diff
@@ -285,26 +232,81 @@ def interpolate_1d(
             if inds.dtype == np.object_:
                 inds = lib.maybe_convert_objects(inds)
 
-    if method in NP_METHODS:
-        # np.interp requires sorted X values, #21037
-        indexer = np.argsort(inds[valid])
-        result[invalid] = np.interp(
-            inds[invalid], inds[valid][indexer], yvalues[valid][indexer]
-        )
-    else:
-        result[invalid] = _interpolate_scipy_wrapper(
-            inds[valid],
-            yvalues[valid],
-            inds[invalid],
-            method=method,
-            fill_value=fill_value,
-            bounds_error=bounds_error,
-            order=order,
-            **kwargs,
-        )
+    def func(yvalues: np.ndarray) -> np.ndarray:
```

Comment:
can you make this a module level function and give it a nice name

Reply:
doc-string as well

Reply:
it needs to be a closure

Reply:
really? can you pass any needed arguments, this makes it really hard to grok

Reply:
the code in this function is applied columnwise, the perf gains will come from doing some things once instead of 100x for the asv.

I prefer this functional approach to pre-validation as it keeps related code together. There is some more cleaning which could move more into the outer function once it's called less often

Reply:
I understand, but having this an outer function with passing args should not have any impact on performance. It simply a more understandable approach. ok with merging then refactoring to be simpler later though.

Reply:
its basically the processing of the index which only needs to be done once. However, with the bulk of the time on the preserve_nans set logic, there is no sig perf gains yet, hence draft for now.

originally grouped the validation see #34628. So can do it that way if preferred.

i.e. method, xvalues = missing.clean_interp_method(method, index, **kwargs)

Reply:
kk sure, i think you have 1 PR pending that we should merge before this i think. but lmk what you prefer.

Reply:
yeah, trying not to affect the work by @cchwala, I have an alternative implementation of limit_direction that does not re-use the preserve_nans set logic, see #34628 

so i'm now happy #34727 need not affect that, but still need to look at the max_gap algo

Reply:
sorry #34749 not #34628